pytorch_lightning/strategies/bagua.py

```diff
@@ -0,0 +1,235 @@
+        reset_seed()
+
+        # determine which process we are and world size
+        self.set_world_ranks()
+
+        self._init_bagua_distributed()
+
+    def _init_bagua_distributed(self) -> None:
+        self._set_node_environment_variables()
+        log.info(
+            "Initializing Bagua Distributed: "
+            f"GLOBAL_RANK: {self.global_rank}, "
+            f"MEMBER: {self.global_rank + 1}/{self.world_size}"
+        )
+
+        # need to set device first before initialize Bagua distributed environment
+        # Note: setup_environment calls super().setup_distributed after calling init_distributed()
+        torch.cuda.set_device(self.local_rank)
+
+        if not is_initialized():
+            bagua.init_process_group()
+
+    def _set_node_environment_variables(self) -> None:
+        """Set the environment variables as required by the :func:`bagua.init_process_group` call.
+
+        This enables the use of other cluster environments which don't set these exact variables, e.g., Bagua can be
+        launched with ``torch.distributed.run``.
+        """
+        os.environ["MASTER_ADDR"] = self.cluster_environment.main_address  # type: ignore[union-attr]
+        os.environ["MASTER_PORT"] = str(self.cluster_environment.main_port)  # type: ignore[union-attr]
+        os.environ["RANK"] = str(self.global_rank)
+        os.environ["NODE_RANK"] = str(self.node_rank)
+        os.environ["WORLD_SIZE"] = str(self.world_size)
+        os.environ["LOCAL_RANK"] = str(self.local_rank)
+
+    def _check_qadam_optimizer(self) -> None:
+        has_qadam_optimizer = any([isinstance(opt, QAdamOptimizer) for opt in self.optimizers])
+
+        if not has_qadam_optimizer or len(self.optimizers) > 1 or len(self.lr_scheduler_configs) > 1:
+            raise MisconfigurationException("Bagua QAdam can only accept one QAdamOptimizer and one LR Scheduler.")
+
+        self._bagua_kwargs["q_adam_optimizer"] = self.optimizers[0]
+
+    def configure_ddp(self) -> None:
+        model = LightningBaguaModule(self.model)  # type: ignore[arg-type]
+        self._model = self._setup_model(model)
+
+        # start the background communication for async algorithm
+        if self.lightning_module.trainer.training and self._bagua_algorithm == "async":
+            self.model.bagua_algorithm.resume(self.model)  # type: ignore
+
+    def _setup_model(self, model: Module) -> BaguaDistributedDataParallel:
+        """Wraps the model into a Bagua distributed module."""
+
+        if self._bagua_algorithm == "qadam":
+            self._check_qadam_optimizer()
+
+        algorithm = Algorithm.init(self._bagua_algorithm, **self._bagua_kwargs)
+        return BaguaDistributedDataParallel(
+            module=model,
+            optimizers=self.optimizers,
+            algorithm=algorithm,
+            gradient_as_bucket_view=self._bagua_flatten,
+        )
+
+    @classmethod
+    def register_plugins(cls, plugin_registry: Dict) -> None:
```

Comment:
This method name has to be register_strategies, otherwise won't be called 
https://github.com/PyTorchLightning/pytorch-lightning/blob/1203094a201bd38f0b8b77d93bc39fc95f06d8ae/pytorch_lightning/strategies/strategy_registry.py#L137

There is no test covers strategy="bagua", so this issue didn't get caught
I will fix it in #11448, or feel free to open a seperate PR to fix this